### PR TITLE
Add directive to define domain for container name

### PR DESCRIPTION
**Pull request by sashaaro**
Container name is required indicator. Will be convenient to have access by container name as subdomain.
I rename `domain` directive to `hostname_domain`. `domain` work as domain for container's names

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,7 +49,8 @@ Example
 
     .:15353 {
         docker unix:///var/run/docker.sock {
-            domain docker.local
+            domain docker.loc
```
Comment (kevinjqiu) [inline README.md]: Is that a typo? `.loc` vs `.local`

**Comment by kevinjqiu** [comment]
Hey @sashaaro  I don't quite understand this change.  Is there a case where `hostname_domain` and `domain` would be set differently?  Could you give an example where both are configured, and when a container is launched, what DNS names would it get?

Thanks!

File: README.md (hunk 1 above)
Comment (sashaaro) [reply README.md]: Probably cause of https://github.com/laravel/valet/issues/566#issuecomment-380421645

**Comment by sashaaro** [comment]
For example 
```
docker unix:///var/run/docker.sock {
        domain docker.loc
    }

docker run -d --name my-portainer -v /var/run/docker.sock:/var/run/docker.sock -v portainer_data:/data portainer/portainer
```
Would be resolved by `my-portainer.docker.loc`
Same for `hostname_domain` directive but it use `--hostname` instead of `--name` for subdomin

```
docker unix:///var/run/docker.sock {
        hostname_domain docker-host.loc
    }

docker run -d --hostname my-portainer -v /var/run/docker.sock:/var/run/docker.sock -v portainer_data:/data portainer/portainer
```
Would be resolved by `my-portainer.docker-host.loc`

It is not clearly from readme file?!

**Comment by kevinjqiu** [comment]
Thanks for the explanation.

**Comment by sashaaro** [comment]
Want to readme would be clear for everyone

**Pull request merged**